sdk/src/androidTest/java/ly/count/android/sdk/ModuleViewsTests.java

```diff
@@ -1439,5 +1440,101 @@ public void internalLimits_setGlobalSegmentation_maxSegmentationValues_interface
     }
     //test for sessions when consent removed
 
+    /**
+     * global seg : avu=4, avi=v1 after truncation: av=v1
+     * Test the truncation of view name and segmentation keys
+     * key length: 2
+     * Global view segmentation values will be truncated and merged to one because they have same start
+     * View name also will be truncated to expected name "VI"
+     * On stop view, global segmentation will not be overridden by the given segmentation after truncation
+     * on stop seg: satalite=hoho, avu=25 after truncation: sa=hoho, av=v1
+     * None of the countly view segmentation keys will be truncated nor overridden
+     */
+    @Test
+    public void internalLimit_recordViewsWithSegmentation() throws JSONException {
+        Map<String, Object> globalSegm = new ConcurrentHashMap<>();
+        globalSegm.put("avu", 4);
+        globalSegm.put("avi", "v1");
+
+        @NonNull CountlyConfig cc = TestUtils.createViewCountlyConfig(false, false, false, safeViewIDGenerator, globalSegm);
+        cc.sdkInternalLimits.setMaxKeyLength(2);
+        cc.setEventQueueSizeToSend(1);
+        Countly mCountly = new Countly().init(cc);
+
+        Map<String, Object> givenStartSegm = new ConcurrentHashMap<>();
+        givenStartSegm.put("sop", 4);
+        String viewID = mCountly.views().startView("VIEW", givenStartSegm);
+
+        Map<String, Object> expectedSegm = new ConcurrentHashMap<>();
+        ClearFillSegmentationViewStart(expectedSegm, "VI", true);
+        expectedSegm.putAll(TestUtils.map("av", "v1", "so", 4));
+
+        ModuleEventsTests.validateEventInRQ(ModuleViews.VIEW_EVENT_KEY, expectedSegm, 0);
+
+        mCountly.views().setGlobalViewSegmentation(TestUtils.map("sunburn", true, "sunflower", "huh"));
+
+        Map<String, Object> endSegm = new ConcurrentHashMap<>();
+        endSegm.put("satellite", "hoho");
+        endSegm.put("avu", 25);
+        mCountly.views().stopViewWithID(viewID, endSegm);
+        ClearFillSegmentationViewEnd(expectedSegm, "VI", null);
+        expectedSegm.putAll(TestUtils.map("av", 25, "sa", "hoho", "su", "huh"));
+
+        ModuleEventsTests.validateEventInRQ(ModuleViews.VIEW_EVENT_KEY, expectedSegm, 1);
+    }
+
+    /**
```

Comment:
codacy says that this comment is really huge,  I also think that it is. An idea to shorten it would be combining lines 1488-1491 into 1 line. line 1497 also seems like it shouldn't be separated

Reply:
It is for readability, we can also omit that codacy rule.